packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -406,7 +411,13 @@ function updateForwardRef(
   let hasId;
   prepareToReadContext(workInProgress, renderLanes);
   if (enableSchedulingProfiler) {
-    markComponentRenderStarted(workInProgress);
+    markComponentRenderStartedInDevTools(workInProgress);
+  }
+  if (enableTracingHooks) {
```

Comment:
The API `markComponentRenderStartedInTracingHooks` already checks `enableTracingHooks`. Should we remove that in the API?

Reply:
I've seen that pattern (having the check both in the caller and in the implementation) in the code for DevTools, so I'm assuming it's probably a safety mechanism (in case you forget the condition in the caller, the only "problem" is you'd have a function call that doesn't do anything).